packages/webapp/src/components/RepeatCropPlan/index.jsx

```diff
@@ -0,0 +1,378 @@
+    defaultValues: {
+      crop_plan_name: fromCreation ? cropPlan.name : '',
+      plan_start_date: origStartDate,
+      repeat_frequency: 1,
+      repeat_interval: { value: 'week', label: t('REPEAT_PLAN.INTERVAL.WEEK') },
+      days_of_week: [getWeekday(origStartDate)],
+      finish: 'after',
+      after_occurrences: 3,
+    },
+  });
+
+  const { historyCancel } = useHookFormPersist(getValues);
+
+  const intervalOptions = [
+    { value: 'day', label: t('REPEAT_PLAN.INTERVAL.DAY') },
+    { value: 'week', label: t('REPEAT_PLAN.INTERVAL.WEEK') },
+    { value: 'month', label: t('REPEAT_PLAN.INTERVAL.MONTH') },
+    { value: 'year', label: t('REPEAT_PLAN.INTERVAL.YEAR') },
+  ];
+
+  const [monthlyOptions, setMonthlyOptions] = useState([]);
+
+  const totalCount = useRef();
+
+  const planStartDate = watch('plan_start_date');
+  const repeatFrequency = watch('repeat_frequency');
+  const repeatInterval = watch('repeat_interval');
+  const daysOfWeek = watch('days_of_week');
+  const monthRepeatOn = watch('month_repeat_on');
+  const finish = watch('finish');
+  const finishOnDate = watch('finish_on_date');
+
+  // Populate monthly options React Select
+  useEffect(() => {
+    if (repeatInterval.value !== 'month') {
+      return;
+    }
+    // Clear when planStartDate is cleared
+    if (!planStartDate) {
+      setMonthlyOptions([]);
+      setValue('month_repeat_on', null);
+      return;
+    }
+
+    const getAndSetMonthlyOptions = async () => {
+      // Store which pattern is currently selected
+      const currentSelection = monthlyOptions?.findIndex(
+        (option) =>
+          // e.g. {"value":8,"label":"every month on the 8th"}
+          JSON.stringify(option) === JSON.stringify(monthRepeatOn),
+      );
+
+      // Utility function uses rrule to generate natural language strings
+      const options = await calculateMonthlyOptions(planStartDate, repeatFrequency);
+
+      setMonthlyOptions(options);
+
+      // Persist original selection
+      if (options[currentSelection]) {
+        setValue('month_repeat_on', options[currentSelection]);
+      }
+    };
+    getAndSetMonthlyOptions();
+  }, [planStartDate, repeatFrequency, repeatInterval]);
+
+  // Count occurences to given "finish on" date
+  useEffect(() => {
+    if (!repeatFrequency || !repeatInterval || !planStartDate || !finishOnDate || finish !== 'on') {
+      return;
+    }
+
+    const count = countOccurrences({
+      planStartDate,
+      repeatFrequency,
+      repeatInterval,
+      daysOfWeek: daysOfWeek ?? [getWeekday(planStartDate)],
+      monthRepeatOn: monthRepeatOn ?? { value: getDate(planStartDate) },
+      finishOnDate,
+      finish,
+    });
+
+    totalCount.current = count;
+
+    trigger('finish_on_date');
+  }, [planStartDate, repeatFrequency, repeatInterval, finishOnDate, monthRepeatOn]);
+
+  // Custom validation for "finish on" date input
+  const validateFinishOnDate = (value) => {
+    if (getValues('finish') === 'after') {
```

Comment:
Is this how you get the latest value?
`const finish = watch('finish');` This can't be used??

Reply:
Correct. Right now the watch value can't be used for the same reason as above (the validation does not clear on interaction with the `finish` radio group if you use the watch value).

But this is only because of how I set up my repetition count useEffect to return when the "finish after _x_  repetitions" is selected... otherwise there is a `trigger('finish_on_date')` within that useEffect which would be sufficient, so depending on when/how the count useEffect is run, this could be changed!